File: Service/OrderInvoiceService/Models/API/Products/Requests/AttributeValuesForProductRequest.cs
// ***********************************************************************
// Assembly         : OrderInvoiceService
// Author           : Admin
// Created          : 02-22-2021
//
// Last Modified By : Admin
// Last Modified On : 02-22-2021
// ***********************************************************************
// <copyright file="AttributeValuesForProductRequest.cs" company="OrderInvoiceService">
//     Copyright (c) . All rights reserved.
// </copyright>
// <summary></summary>
// ***********************************************************************
namespace OrderInvoiceService.Models.API.Products.Requests
{
    /// <summary>
    /// Class AttributeValuesForProductRequest.
    /// </summary>
    public class AttributeValuesForProductRequest
    {
        /// <summary>
        /// Gets or sets the product.
        /// </summary>
        /// <value>The product.</value>
        public ProductModel product { get; set; }
        /// <summary>
        /// Gets or sets the attribute.
        /// </summary>
        /// <value>The attribute.</value>
        public AttributeModel attribute { get; set; }
        /// <summary>
        /// Gets or sets the attribute values.
        /// </summary>
        /// <value>The attribute values.</value>
        public AttributeValueModel[] attributeValues { get; set; }
    }
}

File: DevUIArea/angular-product-setup/src/commonComponents/models/requests/products/attributeValuesForProductRequest.ts
import { ProductAttributeValueModel } from '../../products/productAttributeValueModel';
import { AttributeModel } from '../../products/attributeModel';
import { ProductModel } from '../../products/productModel';


export class AttributeValuesForProductRequest {
    public product: ProductModel;
    public attribute: AttributeModel;
    public attributeValues: ProductAttributeValueModel[];

    constructor() {
        this.product = new ProductModel();
        this.attribute = new AttributeModel();
        this.attributeValues = new Array<ProductAttributeValueModel>()
    }
}